### PR TITLE
[StructuralMechanics] Making MassCalcProcess MPI-Compatible

### PR DESCRIPTION
**Pull request by philbucher**
See title

In addition I save the mass of the element in the nodes, by distributing the the elemental-mass
I need this for some stuff but I think it can also be handy for others
@KratosMultiphysics/structural-mechanics are there any objections to this?

@jcotela I am putting you in review in case you see sth I missed regarding the parallelism

### CODE REVIEW

File: applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp [hunk 1]
```diff
@@ -61,36 +65,45 @@ void TotalStructuralMassProcess::Execute()
...
-            total_mass += density * thickness * volume;
+            element_mass = density * thickness * volume;
         }
 
-        // We restore the current configuration
-        for (std::size_t i_node = 0; i_node < number_of_nodes; ++i_node)
+        // We restore the current configuration and compute the nodal mass
+        for (std::size_t i_node = 0; i_node < number_of_nodes; ++i_node) {
             noalias(r_this_geometry[i_node].Coordinates()) = current_coordinates[i_node];
+            r_this_geometry[i_node].GetValue(NODAL_MASS) += element_mass / number_of_nodes;
```
Comment (jcotela) [inline applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp]: Be careful here, if the node is not locked, several elements might try to access the same node at the same time from different OpenMP threads.

**Review by jcotela — CHANGES_REQUESTED** [review]
I believe this contains an OpenMP race condition

File: applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp [hunk 2]
```diff
@@ -61,36 +65,45 @@ void TotalStructuralMassProcess::Execute()
         // We get the values from the condition
         const Properties& this_properties = it_elem->GetProperties();
 
+        element_mass = 0.0; // resetting the element-mass
+
         if (local_space_dimension == 0) { // POINT MASSES
             if (it_elem->Has(NODAL_MASS))
-                total_mass += it_elem->GetValue(NODAL_MASS);
+                element_mass = it_elem->GetValue(NODAL_MASS);
```
Comment (loumalouomega) [inline applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp:70]: I don't get the changes, why we move from += to just = and vice versa?

Comment (philbucher) [reply applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp:70]: bcs now I am computing the mass element-by-element and not summing up the total mass
after computing it element by element I distribute the mass among the nodes and finally sum up the nodal-masses to compute the total mass

File: applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp (hunk 1 above)
Comment (philbucher) [reply applications/StructuralMechanicsApplication/custom_processes/total_structural_mass_process.cpp]: as discussed, I removed the omp now entirely bcs it is too dangerous with the coordinates-manipulations

**Review by loumalouomega — DISMISSED** [review]
OK

**Review by jcotela — DISMISSED** [review]
Removing the blocker

**Comment by philbucher** [comment]
@loumalouomega Now I am saving the results on the elements, I think this is the better solution
=> from elements one can always compute the Nodal Masses, but not vise-versa

**Comment by loumalouomega** [comment]
OK then

**Review by loumalouomega — DISMISSED** [review]

**Comment by philbucher** [comment]
@loumalouomega sorry for the noise, I removed some leftovers

**Review by loumalouomega — APPROVED** [review]

**Review by loumalouomega — APPROVED** [review]

**Pull request merged**